server/src/main/java/com/hedera/block/server/verification/hasher/CommonUtils.java

```diff
@@ -0,0 +1,181 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.block.server.verification.hasher;
+
+import com.hedera.hapi.block.BlockItemUnparsed;
+import com.hedera.hapi.block.stream.BlockProof;
+import com.hedera.pbj.runtime.io.buffer.Bytes;
+import com.swirlds.common.crypto.DigestType;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.nio.ByteBuffer;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+
+/**
+ * Provides common utility methods for hashing and combining hashes.
+ */
+public final class CommonUtils {
+    private CommonUtils() {
+        throw new UnsupportedOperationException("Utility Class");
+    }
+
+    /**
+     * The size of an SHA-384 hash in bytes.
+     */
+    public static final int HASH_SIZE = DigestType.SHA_384.digestLength();
+
+    /**
+     * The tag for the SHA-384 algorithm.
+     */
+    private static final String sha384HashTag = "SHA-384";
+
+    /**
+     * Returns the SHA-384 hash of the given bytes.
+     * @param bytes the bytes to hash
+     * @return the SHA-384 hash of the given bytes
+     */
+    public static Bytes noThrowSha384HashOf(final Bytes bytes) {
+        return Bytes.wrap(noThrowSha384HashOf(bytes.toByteArray()));
+    }
+
+    /**
+     * Returns the SHA-384 hash of the given byte array.
+     * @param byteArray the byte array to hash
+     * @return the SHA-384 hash of the given byte array
+     */
+    public static byte[] noThrowSha384HashOf(final byte[] byteArray) {
+        try {
+            return MessageDigest.getInstance(sha384HashTag).digest(byteArray);
+        } catch (final NoSuchAlgorithmException fatal) {
+            throw new IllegalStateException(fatal);
+        }
+    }
+
+    /**
+     * Returns a {@link MessageDigest} instance for the SHA-384 algorithm, throwing an unchecked exception if the
+     * algorithm is not found.
+     * @return a {@link MessageDigest} instance for the SHA-384 algorithm
+     */
+    public static MessageDigest sha384DigestOrThrow() {
+        try {
+            return MessageDigest.getInstance(DigestType.SHA_384.algorithmName());
+        } catch (final NoSuchAlgorithmException fatal) {
+            throw new IllegalStateException(fatal);
+        }
+    }
+
+    /**
+     * Hashes the given left and right hashes.
+     * @param leftHash the left hash
+     * @param rightHash the right hash
+     * @return the combined hash
+     */
+    public static Bytes combine(@NonNull final Bytes leftHash, @NonNull final Bytes rightHash) {
+        return Bytes.wrap(combine(leftHash.toByteArray(), rightHash.toByteArray()));
+    }
+
+    /**
+     * Hashes the given left and right hashes.
+     * @param leftHash the left hash
+     * @param rightHash the right hash
+     * @return the combined hash
+     */
+    public static byte[] combine(final byte[] leftHash, final byte[] rightHash) {
```

Comment:
could add the non-null annotation to the parameter for the static code analysis

Reply:
✅ Done!